chain/actors/builtin/miner/v0.go

```diff
@@ -165,26 +165,50 @@ func (s *state0) LoadSectorsFromSet(filter *bitfield.BitField, filterOut bool) (
 		return nil, err
 	}
 
-	if filter != nil {
-		var v cbg.Deferred
-		if err := a.ForEach(&v, func(i int64) error {
+	ret := adt0.MakeEmptyArray(s.store)
+	var v cbg.Deferred
```

Comment:
@magik6k urgh, just realised that this undoes the fasterer changes you introduced here: https://github.com/filecoin-project/lotus/pull/3781/commits/b25dd2a00d9c4d346087438d33a8d789a50aa8f0.

But we do need to do some dance manipulating these, since the SectorOnChainInfo changes between actors 0 and 1. Kick the performance implications down the road for now?

Reply:
We can't, this takes 5min+ for bigger miners, and we call this on each epoch, multiple times

Reply:
I have rewritten this to be faster. It should now fetch exactly the sectors requested.